### PR TITLE
Store locality information in hosts files

### PR DESCRIPTION
**Pull request by benesch**
Store the locality of each node in the cluster's hosts file so that
roachperf can set the --locality flag appropriately when booting the
cluster.

See also: cockroachdb/roachperf#22, cockroachlabs/production#537

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -69,7 +70,17 @@ type VM struct {
...
 	PrivateIP string
 	PublicIP  string
-	Zone       string
+	Zone      string
+}
+
+var regionRE = regexp.MustCompile(`(.*[^-])-?[a-z]$`)
+
+func (vm *VM) locality() string {
+	region := "?"
```
Comment (petermattis) [inline cluster.go]: Perhaps region should be the empty string if there is no region? Not as obvious to the reader, but we almost never view this file.

Comment (benesch) [reply cluster.go]: The following isn't a valid locality:

```
--locality=region=,zone=some-zone
```

So: we could either omit it entirely (`--locality=zone=some-zone`), or return an error to the user. Do you have a preference? I have a slight preference for the latter. The more we can assert about our cluster's environment (e.g., "you will _always_ have a region and a zone"), the easier it is to write nightly tests.

Comment (petermattis) [reply cluster.go]: Ok, makes sense. Carry on.

Comment (benesch) [reply cluster.go]: Ok, roachprod now `log.Fatal`s if it can't parse a VM's region. TFTR!

**Pull request merged**